django_filters/fields.py

```diff
@@ -223,7 +232,10 @@ def __iter__(self):
             yield ("", self.field.empty_label)
         if self.field.null_label is not None:
             yield (self.field.null_value, self.field.null_label)
-        yield from self.choices
+        if DJANGO_50:
+            yield from normalize_choices(self.choices)
```

Comment:
Thus we have the `normalize_choices()` call here which will happen lazily when `.choices` is accessed.

Reply:
Alternatively, if you don't want this branch here you could have `normalize_choices = lambda x: x` in the `except ImportError:` block above to make it a no-op.

Reply:
Happy to go that way if you prefer it.

Reply:
Decided it'd only be worth it if it meant we could eliminate the `DJANGO_50` boolean. But that'll be required anyway for the property setter difference in behaviour, so not worth it.